test/widgets/trip_page_builder_test.dart

```diff
@@ -0,0 +1,279 @@
+        DriveFactory().generateFake(
+            id: 2,
+            endTime: DateTime.now().add(const Duration(days: 1)),
+            hideInListView: false,
+            rides: [
+              RideFactory().generateFake(id: 1, rider: NullableParameter(ProfileFactory().generateFake(id: 1)))
+            ]).toJsonForApi(),
+      ];
+      whenRequest(processor,
+              urlMatcher:
+                  equals('/rest/v1/drives?select=%2A&driver_id=eq.${profile.id}&order=start_time.asc.nullslast'))
+          .thenReturnJson(drives);
+      whenRequest(processor, urlMatcher: matches(RegExp(r'/rest/v1/drives.*&id=eq\.1')), methodMatcher: equals('GET'))
+          .thenReturnJson(drives[0]);
+      whenRequest(processor, urlMatcher: matches(RegExp(r'/rest/v1/drives.*&id=eq\.2')), methodMatcher: equals('GET'))
+          .thenReturnJson(drives[1]);
+      await pumpMaterial(tester, const DrivesPage());
+      await tester.pump();
+      expect(find.byType(DriveCard), findsNWidgets(2));
+    });
+
+    testWidgets('does not show Cards that should not be shown', (WidgetTester tester) async {
+      drives = [
+        DriveFactory().generateFake(
+            id: 1,
+            endTime: DateTime.now().subtract(const Duration(days: 1)),
+            hideInListView: false,
+            rides: [
+              RideFactory().generateFake(id: 1, rider: NullableParameter(ProfileFactory().generateFake(id: 1)))
+            ]).toJsonForApi(),
+        DriveFactory().generateFake(
+            id: 2,
+            endTime: DateTime.now().add(const Duration(days: 1)),
+            hideInListView: true,
+            rides: [
+              RideFactory().generateFake(id: 1, rider: NullableParameter(ProfileFactory().generateFake(id: 1)))
+            ]).toJsonForApi(),
+      ];
+      whenRequest(processor,
+              urlMatcher:
+                  equals('/rest/v1/drives?select=%2A&driver_id=eq.${profile.id}&order=start_time.asc.nullslast'))
+          .thenReturnJson(drives);
+      await pumpMaterial(tester, const DrivesPage());
+      await tester.pump();
+      expect(find.byType(DriveCard), findsNothing);
+      verifyRequestNever(processor,
+          urlMatcher: matches(RegExp(r'/rest/v1/drives.*&id=eq\.1')), methodMatcher: equals('GET'));
+      verifyRequestNever(processor,
+          urlMatcher: matches(RegExp(r'/rest/v1/drives.*&id=eq\.2')), methodMatcher: equals('GET'));
+    });
+    testWidgets('shows Nothing if no Drives are there', (WidgetTester tester) async {
+      whenRequest(processor,
+              urlMatcher:
+                  equals('/rest/v1/drives?select=%2A&driver_id=eq.${profile.id}&order=start_time.asc.nullslast'))
+          .thenReturnJson([]);
+      await pumpMaterial(tester, const DrivesPage());
+      await tester.pump();
+      expect(find.byType(DriveCard), findsNothing);
+      expect(find.byKey(const Key('emptyMessage')), findsOneWidget);
+    });
+  });
+
+  group('shows Cards past', () {
```

Comment:
let's call the group "Past Cards" or something, I think only the test cases themselves should contain verbs